crates/bevy_ui/src/widget/text.rs

```diff
@@ -78,19 +98,34 @@ pub fn text_system(
     let mut new_queue = Vec::new();
     let mut query = text_queries.p2();
     for entity in queued_text.entities.drain(..) {
-        if let Ok((text, style, mut calculated_size)) = query.get_mut(entity) {
+        if let Ok((text, style, mut calculated_size, parent)) = query.get_mut(entity) {
+            let mut scale_factor_width = scale_factor;
+            let mut scale_factor_height = scale_factor;
+            if let Some(parent) = parent {
+                if let Ok(style) = parent_query.get(parent.get()) {
+                    if let Val::Percent(percentage) = style.size.width {
+                        scale_factor_width = percentage as f64 / (scale_factor * 100.);
+                    }
+                    if let Val::Percent(percentage) = style.size.height {
+                        scale_factor_height = percentage as f64 / (scale_factor * 100.);
+                    }
+                }
+            }
```

Comment:
Nit: this is mostly a style issue, but I think some of the rightward drift can be avoided here.

Something like:
```rust
let query_parent = |parent: &Parent| parent_query.get(parent.get()).ok();
if let Some(style) = parent.and_then(query_parent) {
    if let Val::Percent(percentage) = style.size.width {
        scale_factor_width = percentage as f64 / (scale_factor * 100.);
    }
    if let Val::Percent(percentage) = style.size.height {
        scale_factor_height = percentage as f64 / (scale_factor * 100.);
    }
}
```

I don't like the `let mut foo = qux; if bar { foo = baz }` pattern, as it's not very rusty, but I don't see a better way of doing what you are doing here, so it's fine :smile:

Reply:
I think this will be much less of an issue once if-let-chaining gets released https://rust-lang.github.io/rfcs/2497-if-let-chains.html